op-batcher/batcher/driver.go

```diff
@@ -151,6 +155,34 @@ func (l *BatchSubmitter) StartBatchSubmitting() error {
 	return nil
 }
 
+// waitForL2Genesis waits for the L2 genesis time to be reached.
+func (l *BatchSubmitter) waitForL2Genesis() error {
+	genesisTime := time.Unix(int64(l.RollupConfig.Genesis.L2Time), 0)
+	now := time.Now()
+	if now.After(genesisTime) {
+		return nil
+	}
+
+	l.Log.Info("Waiting for L2 genesis", "genesisTime", genesisTime)
+
+	// Create a ticker that fires every 30 seconds
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			remaining := time.Until(genesisTime)
+			l.Log.Info("Waiting for L2 genesis", "remainingTime", remaining.Round(time.Second))
+		case <-time.After(time.Until(genesisTime)):
```

Comment:
Instantiating this genesis time trigger outside of the loop would be nice. It stays the same, and that way we don't create one for every tick. Golang fixed GC of this in v1.23, but we're following geth Go versioning, which targets not the latest version, but the one before, for stability and compatibility reasons.

Reply:
good catch, thanks for the advice.